pkg/blockstorage/azure/auth.go

```diff
@@ -1,110 +1,167 @@
 package azure
 
 import (
-	"context"
-
-	"github.com/Azure/go-autorest/autorest/adal"
-	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/kanisterio/kanister/pkg/blockstorage"
 	"github.com/pkg/errors"
 )
 
+const ActiveDirectory = "activeDirectory"
+
 // currently avaialble types: https://docs.microsoft.com/en-us/azure/developer/go/azure-sdk-authorization
 // to be available with azidentity: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#readme-credential-types
 // determine if the combination of creds are client secret creds
 func isClientCredsAvailable(config map[string]string) bool {
-	return (config[blockstorage.AzureTenantID] != "" &&
+	return config[blockstorage.AzureTenantID] != "" &&
 		config[blockstorage.AzureClientID] != "" &&
-		config[blockstorage.AzureClientSecret] != "")
+		config[blockstorage.AzureClientSecret] != ""
 }
 
 // determine if the combination of creds are MSI creds
 func isMSICredsAvailable(config map[string]string) bool {
 	_, clientIDok := config[blockstorage.AzureClientID]
-	return (clientIDok && config[blockstorage.AzureTenantID] == "" &&
-		config[blockstorage.AzureClientSecret] == "")
+	return clientIDok && config[blockstorage.AzureTenantID] == "" &&
+		config[blockstorage.AzureClientSecret] == ""
+}
+
+func isDefaultCredsAvailable(config map[string]string) bool {
+	_, clientIDok := config[blockstorage.AzureClientID]
+	_, tenantIDok := config[blockstorage.AzureTenantID]
+	_, clientSecretOk := config[blockstorage.AzureClientSecret]
+	return !clientIDok && !tenantIDok && !clientSecretOk
+}
+
+type ClientCredentialsConfig struct {
+	ClientID     string
+	ClientSecret string
+	TenantID     string
+	AuxTenants   []string
+	AADEndpoint  string
+	Resource     string
+}
+
+// Defaults to Public Cloud and Resource Manager Endpoint.
+func NewClientCredentialsConfig(clientID string, clientSecret string, tenantID string) ClientCredentialsConfig {
+	return ClientCredentialsConfig{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		TenantID:     tenantID,
+		Resource:     cloud.AzurePublic.Services[cloud.ResourceManager].Endpoint,
+		//Todo: find a replacement for the AADEndpoint in the new azure sdk
+		AADEndpoint: cloud.AzurePublic.Services[ActiveDirectory].Endpoint,
+		//azure.PublicCloud.ActiveDirectoryEndpoint,
+	}
 }
 
 // Public interface to authenticate with different Azure credentials type
 type AzureAuthenticator interface {
 	Authenticate(creds map[string]string) error
+	GetAuthorizer() azcore.TokenCredential
 }
 
 func NewAzureAuthenticator(config map[string]string) (AzureAuthenticator, error) {
+	// NewAzureAuthenticator opens up the possibility to Auth with:
```

Comment:
[nit] the additional auth possibilities comes with azure's `NewDefaultCredentials`

Reply:
With Azure DefaultCredential usage under hold, reverted to existing `newazureauthenticator` code